Makefile

```diff
@@ -65,8 +65,9 @@ goimports: ## Go fmt your code
 	hack/goimports.sh .
 
 .PHONY: generate
-generate: ## generate code (e.g. deepcopy)
-	./hack/update-generated-deep-copies.sh
+generate:
+	$(DOCKER_CMD) go generate ./pkg/... ./cmd/...
+	hack/goimports.sh .
```

Comment:
when i run this target in no docker mode, make will fail because the git diff returns a non zero value. i think we need to adjust how this gets returned when not using containerized builds.

Reply:
@elmiko 
Docker mode is source of truth, that's what gets run in CI.  The output of go's tooling can change from version to version.  It's a mess, so best practice is to use the docker containers for reproducible stuff.

Reply:
got it. i feel like we should just remove the non-container builds if we aren't going to maintain the "NO_DOCKER" option.

Reply:
We need the "NO_DOCKER" option because the CI uses it.  The jobs are already running in the container, so we pass them NO_DOCKER, so we can't get rid of it.  Having Docker in the Makefile is for us to run it locally in the same environment as the CI.

Reply:
yeah it makes sense, i totally get why we use docker builds for stuff.  i'd just like to see these makefiles become more bulletproof for developers. i find the callout to docker specifically annoying because i don't have it, or an alias, on many machines since it's been removed from more recent fedora versions.

and this is not an issue for this PR, i'm just griping more in general now because it seems like some of these makefiles breakdown when you try to run them locally, and i find it to be a pain for devel stuff.

Reply:
@JoelSpeed I don't know.  This is just copying what we're doing on AWS.  https://github.com/openshift/cluster-api-provider-aws/pull/258

Reply:
Ack, and there's no explanation of why that is on the AWS side either 😞